pyatlan/model/assets/asset00.py

```diff
@@ -6372,6 +6424,636 @@ def create(cls, *, asset: Asset, name: str, link: str) -> Link.Attributes:
+        return None if self.attributes is None else self.attributes.data_products
+
+    @data_products.setter
+    def data_products(self, data_products: Optional[list[DataProduct]]):
+        if self.attributes is None:
+            self.attributes = self.Attributes()
+        self.attributes.data_products = data_products
+
+    @property
+    def parent_domain(self) -> Optional[DataDomain]:
+        return None if self.attributes is None else self.attributes.parent_domain
+
+    @parent_domain.setter
+    def parent_domain(self, parent_domain: Optional[DataDomain]):
+        if self.attributes is None:
+            self.attributes = self.Attributes()
+        self.attributes.parent_domain = parent_domain
+
+    @property
+    def sub_domains(self) -> Optional[list[DataDomain]]:
+        return None if self.attributes is None else self.attributes.sub_domains
+
+    @sub_domains.setter
+    def sub_domains(self, sub_domains: Optional[list[DataDomain]]):
+        if self.attributes is None:
+            self.attributes = self.Attributes()
+        self.attributes.sub_domains = sub_domains
+
+    class Attributes(DataMesh.Attributes):
+        domain_theme_hex: Optional[str] = Field(
+            None, description="", alias="domainThemeHex"
+        )
+        data_products: Optional[list[DataProduct]] = Field(
+            None, description="", alias="dataProducts"
+        )  # relationship
+        parent_domain: Optional[DataDomain] = Field(
+            None, description="", alias="parentDomain"
+        )  # relationship
+        sub_domains: Optional[list[DataDomain]] = Field(
+            None, description="", alias="subDomains"
+        )  # relationship
+
+        @classmethod
+        @init_guid
+        def create(
+            cls,
+            *,
+            name: StrictStr,
+            icon: Optional[AtlanIcon] = None,
+            parent_domain: Optional[DataDomain] = None,
+            parent_domain_qualified_name: Optional[StrictStr] = None,
+        ) -> DataDomain.Attributes:
+            validate_required_fields(["name"], [name])
+            mesh_name = to_camel_case(name)
+            qualified_name = f"default/domain/{mesh_name}"
+            # If "qualified name" of the parent domain is specified
+            if parent_domain_qualified_name:
+                parent_domain = DataDomain()
+                parent_domain.unique_attributes = {
+                    "qualifiedName": parent_domain_qualified_name
+                }
+                qualified_name = f"{parent_domain_qualified_name}/domain/{mesh_name}"
+            icon_str = icon.value if icon is not None else None
+            return DataDomain.Attributes(
+                name=name,
+                parent_domain=parent_domain,
+                mesh_slug=mesh_name,
+                mesh_abbreviation=mesh_name,
+                qualified_name=qualified_name,
+                icon=icon_str,
+            )
```

Comment:
I was expecting to see this attributes-level stuff generated, so this is good — but I don't see the attributes-level template that defines this logic for the code generation? (Are you sure you've committed everything as part of this PR?)